data/datasets.py

```
@@ -6,61 +6,63 @@
 
 
 class VQADataset(Dataset):  # Visual Question Answering Dataset
-    def __init__(self, dataset, tokenizer, image_processor):
+    def __init__(self, dataset, tokenizer, image_processor, mp_image_token_length):
         self.dataset = dataset
         self.tokenizer = tokenizer
         self.image_processor = image_processor
+        self.mp_image_token_length = mp_image_token_length
 
     def __len__(self):
         return len(self.dataset)
 
     def __getitem__(self, idx):
         item = self.dataset[idx]
 
-        # Handle image (it's a list)
-        image_data = item['images']
-        if isinstance(image_data, list) and len(image_data) > 0:
-            image = image_data[0]
-        else:
-            image = image_data
+        # Handle images (should be a list)
+        images_data = item['images']
+        if not isinstance(images_data, list):
+            images_data = [images_data]
 
-        # Now process the image
-        if isinstance(image, Image.Image):
-            if image.mode != 'RGB':
-                image = image.convert('RGB')
-            processed_image = self.image_processor(image)
-        else:
-            print(f"Error processing image at index {idx}")
-            # Create empty tensor with right dimensions as fallback
-            processed_image = torch.zeros(
-                3, cfg.VLMConfig.vit_img_size, cfg.VLMConfig.vit_img_size)
+        # Now process the images
+        processed_images = []
+        for image in images_data:
+            if isinstance(image, Image.Image):
+                if image.mode != 'RGB':
+                    image = image.convert('RGB')
+                processed_image = self.image_processor(image)
+                processed_images.append(processed_image)
+            else:
```

I feel like we could remove this `else` at this point, it just ensures a silent fail when something goes wrong, and I feel like its better to have it throw an error